### PR TITLE
[HOTFIX] vk: Fix MTRSX deadlock

### PR DESCRIPTION
**Pull request by kd-11**
A rather interesting deadlock is possible when MTRSX is active:
If there is a recovery attempt from inside the offloader and a cell thread also happens to fault at the same time, the flush method may end up with recursion. The second pass will perform the flush and change the current command buffer object which the first pass is unaware of when unwinding and proceeds to submit and reference the wrong command buffer for the rest of the operation.
This would typically not be fatal, but the swap routines manually increment the command buffer pointer which means that the command buffer having changed would mean there would be a skipped command buffer. Future attempts to wait for it to complete would cause a deadlock.

Fixes https://github.com/RPCS3/rpcs3/issues/7427

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/VKGSRender.cpp [hunk 1]
```diff
@@ -2273,6 +2273,10 @@ void VKGSRender::do_local_task(rsx::FIFO_state state)
 		// NOTE: This may cause graphics corruption due to unsynchronized modification
 		on_invalidate_memory_range(m_offloader_fault_range, m_offloader_fault_cause);
 		m_queue_status.clear(flush_queue_state::deadlock);
+
+		// Abort all other operations, this is likely coming from offloader::sync() and we need to unwind the call stack.
+		// If flush_command_queue is executed at this point, recursion will cause a deadlock will occur.
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKGSRender.cpp]: Wording

Comment (kd-11) [reply rpcs3/Emu/RSX/VK/VKGSRender.cpp]: Maybe suggest something, looks good to me.

Comment (Megamouse) [reply rpcs3/Emu/RSX/VK/VKGSRender.cpp]: ... recursion will cause a deadlock will occur
To
... recursion will cause a deadlock to occur
Or something like that

Comment (kd-11) [reply rpcs3/Emu/RSX/VK/VKGSRender.cpp]: ok, nice catch. ~~msvc grammarly extension needed~~

Comment (Megamouse) [reply rpcs3/Emu/RSX/VK/VKGSRender.cpp]: :man_student:

**Pull request merged**